Control/public/environment/environmentsPage.js

```diff
@@ -101,6 +101,7 @@ const requestsTable = (model, requests) =>
         )),
         h('td', {style: 'text-align: center;'}, item.owner),
         h('td', {style: 'text-align: center;'}, new Date(item.date).toLocaleString()),
+        h('td', {style: 'text-align: center;font-weight: bold;'}, item.failed ? 'FAILED' : 'ONGOING'),
```

Comment:
Instead of ongoing I would propose you add the loading spinner to get another request out of the way;
```suggestion
        h('td', {style: 'text-align: center;font-weight: bold;'}, item.failed ? 'FAILED' : loading(2)),
```

At the top of the file, import the loading spinner as:
`import loading from './../common/loading.js';`

Reply:
One request was to add "state" (ONGOING/FAILED) and another to add timer.